### ISSUE TITLE
Colors only showing up to line 48 (macOs)

### ISSUE THREAD

**Issue by socialmedialabs** [abe33/atom-pigments#369]
As already reported in issue #247 pigments seemst to fail to parse the complete document and hence colors will only be shown up until line number 48, any line beyond line number 48 pigments will not detect any color.

My setup is as follows
- macOS 10.12.6 (16G1036)
- Atom 1.22.1
- Pigments 0.40.2

**Comment by C-Bouthoorn** [comment]
Is it always line 48? What if you change your font-size, or reduce/increase the amount of colours per line or in total? It doesn't make sense for me to be exactly line 48

**Comment by socialmedialabs** [comment]
Good point, didn't even think of that. It differs indeed. However, it seems the color field is getting kinda squeezed away. See the following screencast I made:
https://www.dropbox.com/s/hhrei1fd6xe28b5/atom_pigments.mov?dl=0

**Comment by socialmedialabs** [comment]
I realize it might be useful for you to know the UI and Syntax themes I am using:

UI Theme: One Dark
Syntax Theme: Obsidian Colors

Although I may mention as well that I have cycled through various UI and syntax theme combinations with the same result.

Would it be helpful for you to know the packages I have installed and activated? Is it possible one of those interfers with pigments? I did, however, disable packages one by one and the issue still occured.

**Comment by C-Bouthoorn** [comment]
You could try running Atom in safe mode. For Linux (and Mac too probably), you can do this by doing the following in a terminal:

```bash
# Go to a new temporary folder
cd $(mktemp -d)
# Start atom in this folder
ATOM_HOME="$(pwd)" atom
```

This will start a clean version of Atom. Try installing Pigments the usual way, open a package in which the error occurs, and see if the problem persists. If it's gone, you could try (re)moving your old `ATOM_HOME` and starting with a fresh Atom.

**Comment by socialmedialabs** [comment]
I just did that and it resulted in the exact same behaviour. The colors either get squeezed vertically until invisible or they simply disappear.

**Comment by socialmedialabs** [comment]
In the meantime I have updated to macOS 10.13 and the problem persists. However, I just discovered that it works in `native-background` and `native-outline` mode. These modes, however, are unfortunately useless for me.

**Comment by fndelacruz** [comment]
@socialmedialabs did you ever find a fix for this? I have the same issue.

**Comment by socialmedialabs** [comment]
@fndelacruz no, unfortunately I haven't 😕

**Comment by AJerm** [comment]
I'm also having this issue.

1.24 Atom on MacOS 10.13.3.

You can see if I keep pushing these colours down the page, the second one gets sliced in half at line 58 and no colours render past that.

![image](https://user-images.githubusercontent.com/1643439/36341213-e3a1c430-13e2-11e8-9909-583980ed4f71.png)

**Comment by joansalasoler** [comment]
Seems to fix this issue without side effects:

.pigments-gutter-marker span {
    background-attachment: scroll !important;
}

**Comment by socialmedialabs** [comment]
@joansalasoler nice one! Fixes it for me as well. I added this to my atom styles.less and it works fine.

**Comment by X-Raym** [comment]
I confirm the issue. For me it was line 48.

[Demo](https://i.imgur.com/QoHfwe5.gif)

@joansalasoler many thanks !! this trick works !

@C-Bouthoorn It would be nice if this could be fixed natively though, cause it is a bit an unexpected behavior 😿

**Comment by X-Raym** [comment]
@joansalasoler actually you are missing a class, for when the cursor is on the color word itself:

```css
.pigments-gutter-marker span, .pigments-gutter-marker-in-selection span {
    background-attachment: scroll !important;
}

This is the fully working code :P
```

**Comment by dutsov** [comment]
I had the same issue on Linux Mint 19.1. background and outline was working but was no use for my setup. The @X-Raym fix the issue at once!

**Comment by C-Bouthoorn** [comment]
Please check if #420 fixed the problem

**Comment by krymzn** [comment]
As [mentioned](https://github.com/abe33/atom-pigments/issues/438#issuecomment-592192004) in #438 by @Cxarli, directly pulling from git worked for me. All marker types now show below, in my case, line 40.
